src/Fmt_ast.ml

```diff
@@ -3116,6 +3116,13 @@ and fmt_label_declaration c ctx decl ?(last = false) =
     | `Loose -> true
     | `Tight_decl | `Tight -> false
   in
+  let fmt_semicolon =
```

Comment:
For readability, I suggest to move the formatting of the before case in the same pattern matching. Currently it is done in 2 different ways for records declarations and inline records.

Reply:
I'm started doing that but it requires an extensive rewriting, if think it should be in another PR.

Reply:
Having the semicolon in `fmt_label_declaration` makes possible to have the semicolon before the comment (the comment at the end of the line is attached to the type) :
```ocaml
-    margin: int;  (** Format code to fit within [margin] columns. *)
-    max_iters: int;
+    margin: int  (** Format code to fit within [margin] columns. *);
+    max_iters: int
         (** Fail if output of formatting does not stabilize within
-            [max_iters] iterations. *)
+            [max_iters] iterations. *);
     module_item_spacing: [`Compact | `Sparse];
-    ocp_indent_compat: bool;  (** Try to indent like ocp-indent *)
+    ocp_indent_compat: bool  (** Try to indent like ocp-indent *);
```

Reply:
I agree. I was suggesting moving the semicolon logic inside `fmt_label_declaration`

Reply:
Now the whole semicolon logic is in fmt_label_declaration.